lib/mix/tasks/write_stop_place_ids.ex

```diff
@@ -0,0 +1,159 @@
+defmodule Mix.Tasks.WriteStopPlaceIds do
+  @moduledoc """
+  Creates a `Dotcom.StopPlaceIds` module with a single function `stop_place_id/1`.
+  The function is a simple one to one mapping of stop ids to Google place ids.
+  """
+
+  @shortdoc "Create the `Dotcom.StopPlaceIds` module."
+
+  use Mix.Task
+
+  @file_path "lib/dotcom/stop_place_ids.ex"
+  @google_api_key System.get_env("GOOGLE_API_KEY", "")
+  @google_url "https://places.googleapis.com/v1/places:searchNearby"
+  @included_types [
+    "bus_station",
+    "bus_stop",
+    "ferry_terminal",
+    "subway_station",
+    "train_station",
+    "transit_station"
+  ]
+
+  @impl Mix.Task
+  def run(_) do
+    :ok = write_header()
+
+    :ok = write_place_ids()
+
+    :ok = write_footer()
+  end
+
+  # Given a latitude and longitude, construct a request body for Google.
+  # We don't use the name of the location.
+  # Rather, we use the latitude and longitude, restrict the types of places returned,
+  # order them by distance, and take the first.
+  # This means that accuracy is based on the stop latitude and longitude accuracy
+  # which should be more accurate than trying to match names.
+  defp body(latitude, longitude) do
+    %{
+      "includedTypes" => @included_types,
+      "maxResultCount" => 1,
+      "rankPreference" => "DISTANCE",
+      "locationRestriction" => %{
+        "circle" => %{
+          "center" => %{
+            "latitude" => latitude,
+            "longitude" => longitude
+          },
+          "radius" => 100.0
+        }
+      }
+    }
+    |> :json.encode()
+  end
+
+  # Get an individual place id and match it with its stop id.
+  defp get_place_id({id, latitude, longitude}) do
+    body = body(latitude, longitude)
+    headers = headers()
+
+    # We have to wait so we don't exceed our rate limit of 10 requests per second.
+    :timer.sleep(1000)
+
+    {:ok, {{_, 200, _}, _, response}} =
+      :httpc.request(:post, {@google_url, headers, ~c"application/json", body}, [], [])
+
+    places =
+      response
+      |> Kernel.to_string()
+      |> :json.decode()
```

Comment:
suggestion(non-blocking): using `Req.post/2` would save you the body encoding and response decoding calls...

Reply:
Yeah, but using httpc we don't have to use a nimble pool...which we don't really need for a script.